travis-build.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
+
+# exit when any command fails
+set -e
+
+echo "Building and testing artifacts, and creating pom files"
+./gradlew -s -i --scan build publishToMavenLocal codeCoverageReport
```

Comment:
Could you remove the `-i` option from this command? Since we have logging enabled for our tests, this can produce a ton of output.
I think it should be okay to use `-i` on lines 10 and 17 though.